qualtran/bloqs/arithmetic.py

```diff
@@ -578,6 +578,109 @@ def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
         return {(Add(self.bitsize), 1), (ArbitraryClifford(n=2), self.bitsize)}
 
 
+@frozen
+class HammingWeightCompute(GateWithRegisters):
+    r"""A gate to compute the hamming weight of n-bit register in a new log_{n} bit register.
+
+    Implements $U|x\rangle |0\rangle \rightarrow |x\rangle|hamming_weight(x)\rangle$
```

Comment:
probably want some \mathrm's around the hamming_weight

Reply:
And also check that the `_` renders properly sometimes this has a weird effect

Reply:
Done